src/Tasks/StateFileBase.cs

```diff
@@ -77,8 +77,17 @@ internal static StateFileBase DeserializeCache(string stateFile, TaskLoggingHelp
                     using (FileStream s = File.OpenRead(stateFile))
                     {
                         using var translator = BinaryTranslator.GetReadTranslator(s, buffer: null);
+
                         byte version = 0;
                         translator.Translate(ref version);
+                        // If retVal is still null or the version is wrong, log a message not a warning. This could be a valid cache with the wrong version preventing correct deserialization.
+                        // For the latter case, internals may be unexpectedly null.
+                        if (retVal == null || version != CurrentSerializationVersion)
+                        {
+                            log.LogMessageFromResources("General.CouldNotReadStateFileMessage", stateFile, log.FormatResourceString("General.IncompatibleStateFileType"));
```

Comment:
I think it continues to make sense for this to be a quiet message. But do we have a test in place that this is _not_ the case for freshly-built-everything using the SDK cache?

Reply:
We do not. If I remember correctly, we decided it would be too hard for some reason.

Reply:
Can we at least get it added to the manual SDK tests?

Reply:
Yeah, that sounds like a good idea to me. @marcpopMSFT, agree? If so, I can write up some (very detailed) instructions on testing it.

Reply:
Please reach out to Richa with the scenario you want covered.  The difficulty with fresh machine tests as you can only have one.